process/block/interceptedBlocks/interceptedEquivalentProof.go

```diff
@@ -0,0 +1,170 @@
+	DataBuff          []byte
+	Marshaller        marshal.Marshalizer
+	ShardCoordinator  sharding.Coordinator
+	HeaderSigVerifier consensus.HeaderSigVerifier
+	Headers           dataRetriever.HeadersPool
+}
+
+type interceptedEquivalentProof struct {
+	proof             *block.HeaderProof
+	isForCurrentShard bool
+	headerSigVerifier consensus.HeaderSigVerifier
+	headers           dataRetriever.HeadersPool
+}
+
+// NewInterceptedEquivalentProof returns a new instance of interceptedEquivalentProof
+func NewInterceptedEquivalentProof(args ArgInterceptedEquivalentProof) (*interceptedEquivalentProof, error) {
+	err := checkArgInterceptedEquivalentProof(args)
+	if err != nil {
+		return nil, err
+	}
+
+	equivalentProof, err := createEquivalentProof(args.Marshaller, args.DataBuff)
+	if err != nil {
+		return nil, err
+	}
+
+	return &interceptedEquivalentProof{
+		proof:             equivalentProof,
+		isForCurrentShard: extractIsForCurrentShard(args.ShardCoordinator, equivalentProof),
+		headerSigVerifier: args.HeaderSigVerifier,
+		headers:           args.Headers,
+	}, nil
+}
+
+func checkArgInterceptedEquivalentProof(args ArgInterceptedEquivalentProof) error {
+	if len(args.DataBuff) == 0 {
+		return process.ErrNilBuffer
+	}
+	if check.IfNil(args.Marshaller) {
+		return process.ErrNilMarshalizer
+	}
+	if check.IfNil(args.ShardCoordinator) {
+		return process.ErrNilShardCoordinator
+	}
+	if check.IfNil(args.HeaderSigVerifier) {
+		return process.ErrNilHeaderSigVerifier
+	}
+	if check.IfNil(args.Headers) {
+		return process.ErrNilHeadersDataPool
+	}
+
+	return nil
+}
+
+func createEquivalentProof(marshaller marshal.Marshalizer, buff []byte) (*block.HeaderProof, error) {
+	headerProof := &block.HeaderProof{}
+	err := marshaller.Unmarshal(headerProof, buff)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Trace("interceptedEquivalentProof successfully created",
+		"header hash", logger.DisplayByteSlice(headerProof.HeaderHash),
+		"header shard", headerProof.HeaderShardId,
+		"header epoch", headerProof.HeaderEpoch,
+		"header nonce", headerProof.HeaderNonce,
+		"bitmap", logger.DisplayByteSlice(headerProof.PubKeysBitmap),
+		"signature", logger.DisplayByteSlice(headerProof.AggregatedSignature),
+	)
+
+	return headerProof, nil
+}
+
+func extractIsForCurrentShard(shardCoordinator sharding.Coordinator, equivalentProof *block.HeaderProof) bool {
+	proofShardId := equivalentProof.GetHeaderShardId()
+	if proofShardId == core.MetachainShardId {
+		return true
+	}
+
+	return proofShardId == shardCoordinator.SelfId()
+}
+
+// CheckValidity checks if the received proof is valid
+func (iep *interceptedEquivalentProof) CheckValidity() error {
+	err := iep.integrity()
+	if err != nil {
+		return err
+	}
+
+	hdr, err := iep.headers.GetHeaderByHash(iep.proof.HeaderHash)
+	if err != nil {
+		return err
+	}
+
+	return iep.headerSigVerifier.VerifySignatureForHash(hdr, iep.proof.HeaderHash, iep.proof.PubKeysBitmap, iep.proof.AggregatedSignature)
```

Comment:
I don't think we need the header to verify the equivalent proof (at least not in its final form)
Is the header only used for the randomness? We no longer need it as we don't care if the leader has signed or not, and we have the bitmap which refers to the fixed order.